x-pack/plugins/uptime/public/pages/synthetics/step_detail_page.tsx

```diff
@@ -5,11 +5,132 @@
+import { useUiSetting$ } from '../../../../../../src/plugins/kibana_react/public';
+import { StepPageTitleContent } from '../../components/monitor/synthetics/step_detail/step_page_title';
+import { getJourneySteps } from '../../state/actions/journey';
+
+export const useStepDetailPage = (): {
+  activeStep?: JourneyStep;
+  checkGroup: string;
+  hasNextStep: boolean;
+  hasPreviousStep: boolean;
+  journey?: JourneyState;
+  stepIndex: number;
+} => {
+  const dispatch = useDispatch();
+
+  const { checkGroupId: checkGroup, stepIndex: stepIndexString } = useParams<{
+    checkGroupId: string;
+    stepIndex: string;
+  }>();
+
+  console.log('check group', checkGroup, 'step index', stepIndexString, useParams());
+  useEffect(() => {
+    if (checkGroup) {
+      dispatch(getJourneySteps({ checkGroup, syntheticEventTypes: ['step/end'] }));
+    }
+  }, [dispatch, checkGroup]);
+
+  const stepIndex = Number(stepIndexString);
+  const journeys = useSelector(journeySelector);
+  const journey: JourneyState | null = journeys[checkGroup] ?? null;
+
+  const memoized = useMemo(
+    () => ({
+      hasPreviousStep: stepIndex > 1 ? true : false,
+      activeStep: journey?.steps?.find((step) => step.synthetics?.step?.index === stepIndex),
+      hasNextStep: journey && journey.steps && stepIndex < journey.steps.length ? true : false,
+    }),
+    [journey, stepIndex]
+  );
+
+  return {
+    checkGroup,
+    journey,
+    stepIndex,
+    ...memoized,
+  };
+};
+
+export const StepDetailPageHeader = () => {
+  const { activeStep, journey } = useStepDetailPage();
+  console.log('active step', activeStep);
+  return <>{journey && activeStep && activeStep.synthetics?.step?.name}</>;
+};
+
+export const StepDetailPageRightSideItem = () => {
+  const history = useHistory();
+
+  const [dateFormat] = useUiSetting$<string>('dateFormat');
+
+  const { journey } = useStepDetailPage();
+
+  const handleNextRun = useCallback(() => {
+    history.push(`/journey/${journey?.details?.next?.checkGroup}/step/1`);
+  }, [history, journey?.details?.next?.checkGroup]);
+
+  const handlePreviousRun = useCallback(() => {
+    history.push(`/journey/${journey?.details?.previous?.checkGroup}/step/1`);
+  }, [history, journey?.details?.previous?.checkGroup]);
+
+  if (!journey) return null;
+
+  return (
+    <StepPageNavigation
+      dateFormat={dateFormat}
+      handleNextRun={handleNextRun}
+      handlePreviousRun={handlePreviousRun}
+      nextCheckGroup={journey.details?.next?.checkGroup}
+      previousCheckGroup={journey.details?.previous?.checkGroup}
+      checkTimestamp={journey.details?.timestamp}
+    />
+  );
+};
+
+export const StepDetailPageChildren = () => {
+  const history = useHistory();
+
+  const {
+    activeStep,
+    checkGroup,
+    hasPreviousStep,
+    hasNextStep,
+    journey,
+    stepIndex,
+  } = useStepDetailPage();
+
+  const handleNextStep = useCallback(() => {
```

Comment:
Is there a way to move this function to `useStepDetailPage`, since it's used by both consumers of `useStepDetailPage`?

Reply:
Great feedback. https://github.com/elastic/kibana/pull/104939/commits/97efe098ec0892b474a5e9996167c6c3882656af